community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/LogTailScanner.java

```
@@ -147,28 +174,46 @@ private LogTailInformation update() throws IOException
 
     protected LogTailInformation latestCheckPoint( long fromVersionBackwards, long version,
             LogEntryStart latestStartEntry, long oldestVersionFound, CheckPoint latestCheckPoint,
-            LogEntryVersion latestLogEntryVersion ) throws IOException
+            LogEntryVersion latestLogEntryVersion, boolean commitsAfterCheckPoint ) throws IOException
     {
         // Is the latest start entry in this log file version later than what the latest check point targets?
         LogPosition target = latestCheckPoint.getLogPosition();
-        boolean startEntryAfterCheckPoint = latestStartEntry != null &&
-                latestStartEntry.getStartPosition().compareTo( target ) >= 0;
+        boolean startEntryAfterCheckPoint = ((latestStartEntry != null) &&
+                (latestStartEntry.getStartPosition().compareTo( target ) >= 0)) ||
+                commitsAfterCheckPoint;
         if ( !startEntryAfterCheckPoint )
         {
             if ( target.getLogVersion() < version )
             {
                 // This check point entry targets a previous log file.
                 // Go there and see if there's a transaction. Reader is capped to that log version.
-                startEntryAfterCheckPoint = extractFirstTxIdAfterPosition( target, version ) !=
-                        LogTailInformation.NO_TRANSACTION_ID;
+                try
+                {
+                    startEntryAfterCheckPoint =
+                            extractFirstTxIdAfterPosition( target, version ) != LogTailInformation.NO_TRANSACTION_ID;
+                }
+                catch ( LogReadingException lre )
+                {
+                    startEntryAfterCheckPoint = true;
```

Should we silently accept such failure always, or only on "force"? This is changing default behaviour isn't it?